### PR TITLE
(PUP-6660) Remove `--no-use_cached_catalog` flag from `--test` flag

### PR DESCRIPTION
**Pull request by Magisus**
This reverts commit b330957e6ae2c6af0701c696cba628a574e7683f.

This reverts the change that added the `--no-use_cached_catalog` flag to
`--test`, because this made it impossible to ever use the cached catalog
during a test run. The user can once again decide whether a test
run should attempt to use cached catalogs or not.

It also reverts 3c8e953fcc02bfc98db40d87514c5e4b182cca44, which added an
associated test.

### CODE REVIEW

File: acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb [hunk 1]
```diff
@@ -117,8 +117,7 @@
 MANIFEST
 
       step "Run agent again using --use_cached_catalog and ensure content from the first code_id is used"
-      on(agent, puppet("agent", "--no-usecacheonfailure", "--no-splay", "--show_diff", "--no-daemonize", 
-                       "--verbose", "--onetime", "--detailed-exitcodes", "--use_cached_catalog", "--server #{master}"), :acceptable_exit_codes => [0,2])
+      on(agent, puppet("agent", "-t", "--use_cached_catalog", "--server #{master}"), :acceptable_exit_codes => [0,2])
```
Comment (Magisus) [inline acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb]: How did this test ever pass if adding `--use_cached_catalog` did not override what was in `--test`...?

**Comment by joshcooper** [comment]
nit, typo in the commit message `cahced catalog`

**Comment by Magisus** [comment]
Fixed.

**Comment by joshcooper** [comment]
Verified `--test` no longer implies `use_cached_catalog`:

```
$ bx puppet agent -t --server chara
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for chara.corp.puppetlabs.net
Info: Applying configuration version '1494612278'
Notice: Applied catalog in 0.04 seconds
```

But it can be requested:

```
$ bx puppet agent -t --server chara --use_cached_catalog
Info: Using cached catalog from environment 'production'
Info: Loading facts
Info: Applying configuration version '1494612278'
Notice: Applied catalog in 0.04 seconds
```

It can also be set to false to prevent the agent from falling back to the cached catalog:

```
$ bx puppet agent -t --server chara --no-use_cached_catalog
Warning: Unable to fetch my node definition, but the agent run will continue:
Warning: Failed to open TCP connection to chara:8140 (Connection refused - connect(2) for "chara" port 8140)
...
Error: Could not retrieve catalog from remote server: Failed to open TCP connection to chara:8140 (Connection refused - connect(2) for "chara" port 8140)
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
...
```

**Pull request merged**